### PR TITLE
WFLY-6724/JBEAP-5029 test for clustered JPA 2LC

### PR DESCRIPTION
**Pull request by jmartisk**
https://issues.jboss.org/browse/WFLY-6724

### CODE REVIEW

File: testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/jpa2lc/ClusteredJPA2LCTestCase.java [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+        final ModelNode result1 = client1.execute(createEntityReplicatedCacheOp);
+        Assert.assertTrue(result1.toJSONString(false), result1.get(OUTCOME).asString().equals(SUCCESS));
+
+        NodeUtil.deploy(this.deployer, DEPLOYMENTS);
+    }
+
+    /**
+     * We have a replicated entity cache between two nodes. Cache an entity on node0 and then verify that
+     * node1 sees the cached entity as well. Then try evicting it on one node and see if it's been evicted
```
Comment (scottmarlow) [inline testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/jpa2lc/ClusteredJPA2LCTestCase.java]: Nice to see comments added here, that describe what is expected!  Could you add a note as well that mentions whether the eviction occurs synchronously as part of the application (JTA) transaction or asynchronously?

Comment (jmartisk) [inline testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/jpa2lc/ClusteredJPA2LCTestCase.java]: Yeah I can add that. Actually that sounds like an idea for another test to add - where eviction would be done with TransactionAttributeType.NEVER. Does that make sense? Not sure how safe that is though, can't it cause intermittent failures due to race conditions or something?

Comment (scottmarlow) [inline testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/jpa2lc/ClusteredJPA2LCTestCase.java]: The comments in [StatefulWithXPCFailoverTestCase](https://github.com/wildfly/wildfly/tree/master/testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/ejb/xpc) are terrible (hardly any) but that test uses an (JPA) extended persistence context without an active transaction, to verify that pending database changes (referenced by the expended persistence context until the next active transaction) are replicated between nodes.  

I'm not really sure what the answer to your question is about testing eviction with TransactionAttributeType.NEVER.  Years ago, I think that the second level cache would be updated immediately, but I'm not sure currently if that happens immediately or after the current component invocation completes.

Comment (jmartisk) [inline testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/jpa2lc/ClusteredJPA2LCTestCase.java]: Ok let's leave it now, maybe later. I just updated the comment as you suggested.

File: testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/jpa2lc/ClusteredJPA2LCTestCase.java [hunk 2]
```diff
@@ -0,0 +1,229 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2012, Red Hat, Inc., and individual contributors
```
Comment (rhusar) [inline testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/jpa2lc/ClusteredJPA2LCTestCase.java]: 2016

Comment (jmartisk) [inline testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/jpa2lc/ClusteredJPA2LCTestCase.java]: mmmmhm, okay, fixed

**Pull request merged**